nixos/modules/installer/netboot/kexec-boot.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+set -e
+
+kexecTree=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+initrd=$kexecTree/initrd
+kernelParams=$(<"$kexecTree/kernel-params")
+
+usage() {
+    printf >&2 'usage: %s [%s] [%s]\n' "$0" \
+        '--ssh-authorized-keys <file>' \
+        '--append <cmdline>'
+    exit 2
+}
+
+while (( $# )); do
+    arg=$1; shift 1
+    case "$arg" in
+        --ssh-authorized-keys)
+            (( $# )) || usage
+            authorizedKeysFile=$1; shift 1
+            ;;
+        --append)
+            (( $# )) || usage
+            kernelParams="$kernelParams $1"; shift 1
+            ;;
+        *)
+            usage
+            ;;
+    esac
+done
+
+if ! kexec --version &>/dev/null; then
+    printf >&2 'kexec not found: please install kexec-tools\n'
+    exit 1
+fi
+
+if (( EUID != 0 )); then
+    printf >&2 'kexec-boot must be run as root\n'
+    exit 1
+fi
+
+cleanup() {
+    if [[ -v workDir ]]; then
+        rm -rf "$workDir"
+    fi
+}
+
+trap cleanup EXIT
+
+if [[ -v authorizedKeysFile ]]; then
+    workDir=$(mktemp -d --tmpdir nixos-kexec-initrd.XXXXXXXXXX)
+    keysDir=$workDir/overlay/etc/ssh/authorized_keys.d
+    mkdir -p -- "$keysDir"
+    install --mode=0644 -- "$authorizedKeysFile" "$keysDir/nixos"
+    {
+        (
+            cd -- "$workDir"
+            find overlay -print0 | cpio -o -H newc --null --quiet
```

Comment:
I think I actually had issues with `cpio` in some linux distributions to not being able to properly pack directories. I think it were some redhat based ones and or debian ones.

Reply:
Hmm, did you maybe give it a directory name without the entire tree of file names inside? I'd be surprised if `cpio` was that nonfunctional on any major distro, since it'll usually be used for assembling the native initramfs. For what it's worth I tested this on Ubuntu 22.04.

Reply:
I think newer versions also were fine but than you quite often have old distros for a while for longer than you want.

Reply:
I suppose bundling our own cpio would solve this. I'd be curious to know if you remember what distribution had this problem though, as I wonder how you would even use a `cpio(1)` that broken for anything; even the most simple initramfs involves a few directories.